plugins/steve/resource-class.ts

```diff
@@ -32,8 +32,29 @@ import {
   AS, _YAML, MODE, _CLONE, _EDIT, _VIEW, _UNFLAG, _CONFIG
 } from '@/config/query-params';
 
+// eslint-disable-next-line
```

Comment:
Eslint was complaining A LOT about the order of imports but I couldn't figure out any order that it would consider to be valid.

Reply:
I have been forced to add this:
```js
    'import/order':                         'off',
    'import/no-named-as-default':           'off',
```

Reply:
There was a vs code extension that helps with ordering imports called `Typescript Hero`. It's great but enabling it on a project with unsorted imports leads to lots of churn when creating new PRs. We could start this process by encouraging developers to enable it (creates churn) or spend about half hour manually going through each file and fixing (automated tools making mass changes...). I think the former is preferable, any thoughts?

Reply:
FWIW In terms of rules I previously created an issue with some of the more problematic one's that were disabled during the first TS migration phase - https://github.com/rancher/dashboard/issues/3563. Most related to Vue/JS though

Reply:
Oh this is something I forgot to add in the list :)
We should definitely integrate some automation with:
- `husky`, to run scripts on pre-commit, pre-push and commit message
-`lint-stage`, to target the scripts only for the staged files

I am not entirely sure whether is possible to integrate that plugin, but we may find an analogue library/script for such duty.

Reply:
> We could start this process by encouraging developers to enable it (creates churn) or spend about half hour manually going through each file 

I agree that the former is preferable. Ideally the extension's changes would be in a separate commit.

Reply:
@cnotv I don't understand - why should we use those scripts?

Reply:
TypeScript Hero didn't work for me but I tried "TypeScript Import Sorter" and it worked.